src/table/table-role/use-grid-navigation.ts

```diff
@@ -184,16 +201,20 @@ class GridNavigationModel {
   };
 
   private onTableNodeMutation = (mutationRecords: MutationRecord[]) => {
+    // When focused cell is un-mounted the focusout event handler removes this.cell,
+    // while this.prevFocusedCell is retained until the next focusin event.
     if (!this.prevFocusedCell) {
       return;
     }
 
-    // When focused cell was un-mounted - re-apply focus to the same location.
+    // The lost focus in an unmount event is reapplied to the table using the previous cell position.
+    // The moveFocusBy takes care of finding the closest position if the previous one no longer exists.
     for (const record of mutationRecords) {
       if (record.type === 'childList') {
         for (const removedNode of Array.from(record.removedNodes)) {
```

Comment:
Why do we need to observe removed nodes? When a currently focused node gets removed, the table cell should receive a blur event, right?

Reply:
It does, but the blur event also fires, say, when the user clicks somewhere else in which case we do not want to restore the focus.